crates/backend/src/scheduler.rs

```diff
@@ -0,0 +1,93 @@
+use chrono::{DateTime, Local, NaiveDateTime, NaiveTime, TimeZone, Utc};
+use dbus::notification::ScheduledNotification;
+use std::cmp::Reverse;
+use std::collections::BinaryHeap;
+
+pub struct Scheduler {
+    queue: BinaryHeap<Reverse<ScheduledNotification>>,
+}
+
+impl Scheduler {
+    pub fn new() -> Self {
+        Scheduler {
+            queue: BinaryHeap::new(),
+        }
+    }
+    pub fn add(&mut self, notification: ScheduledNotification) {
+        match Self::parse_time(&notification.time) {
+            Ok(parsed_time) => {
+                dbg!(&parsed_time);
+
+                let scheduled_notification = ScheduledNotification {
+                    time: parsed_time.to_rfc3339(),
+                    data: notification.data,
+                    id: notification.id,
+                };
+                self.queue.push(Reverse(scheduled_notification));
+            }
+            Err(e) => {
+                eprintln!("Error parsing time '{}': {}", notification.time, e);
+            }
+        }
+    }
+
+    fn parse_time(time_str: &str) -> Result<DateTime<Utc>, chrono::ParseError> {
+        let now = Utc::now();
+
+        if let Ok(duration) = humantime::parse_duration(time_str) {
+            return Ok(now + chrono::Duration::from_std(duration).unwrap());
+        }
+
+        let datetime_formats = [
+            "%d.%m.%Y %H:%M",    // 01.01.2025 00:00
+            "%Y-%m-%d %H:%M",    // 01-01-2025 00:00
+            "%d.%m.%Y %I:%M %p", // 01.01.2025 12:00 AM
+            "%Y-%m-%d %I:%M %p", // 01-01-2025 12:00 AM
+        ];
```

Comment:
The second and last are not matched to examples in commentaries. The second will match only "2025-01-01 00:00" and the last one will match only "2025-01-01 12:00 AM".